### PR TITLE
feat: Golang Support for stack analyses

### PR DESCRIPTION
**Pull request by dgpatelgit**
Added support for GO ecosystem:

https://issues.redhat.com/browse/APPAI-1462

### CODE REVIEW

**Review by arajkumar — COMMENTED** [review]
Can we refactor the code to follow "Open-close" principle? The existing one is too invasive and needs modification in quite a bit of places in the form of `if..else` which is against "Open-close" principle.

My proposal is to come up with an interface and each ecosystem can implement the interface to generate the dependency graph?

Something like below,

```
interface IDependencyGraphGenerator {
    async resolvedGraph(manifestPath: uri): uri;
}

// for each ecosystem
// pypi.ts
class PyPi implements IDependencyGraphGenerator {
    async resolvedGraph(manifestPath: uri): uri {
        ..
        return exec(command)
    }
}

// golang.ts
class Golang implements IDependencyGraphGenerator {
}
...
```

**Comment by dgpatelgit** [comment]
> Can we refactor the code to follow "Open-close" principle? The existing one is too invasive and needs modification in quite a bit of places in the form of `if..else` which is against "Open-close" principle.
> 
> My proposal is to come up with an interface and each ecosystem can implement the interface to generate the dependency graph?
> 
> Something like below,
> 
> ```
> interface IDependencyGraphGenerator {
>     async resolvedGraph(manifestPath: uri): uri;
> }
> 
> // for each ecosystem
> // pypi.ts
> class PyPi implements IDependencyGraphGenerator {
>     async resolvedGraph(manifestPath: uri): uri {
>         ..
>         return exec(command)
>     }
> }
> 
> // golang.ts
> class Golang implements IDependencyGraphGenerator {
> }
> ...
> ```

This pattern is really good to handle the problem we have with various ecosystem. But taking time constraint we have this sprint, can we create a separate story for code refactoring / cleanup, where we can clean up all ecosystem code as well.

**Comment by arajkumar** [comment]
@dgpatelgit It won't take a lot of time to refactor., because the logic is going to remain same. I prefer to do it now rather later.

File: src/Utils.ts [hunk 1]
```diff
@@ -23,4 +23,11 @@ export namespace Utils {
       .get('pythonPath');
     return pypiPath ? `${pypiPath}` : 'python';
   }
+
+  export function getGoExecutable(): string {
+    const goPath: string = vscode.workspace
+      .getConfiguration('go')
+      .get('goPath');
+    return goPath ? `${goPath}` : 'go';
+  }
```
Comment (arajkumar) [inline src/Utils.ts:32]: Isn't vscode settings `go.goPath` referes to [GOPATH](https://golang.org/doc/gopath_code.html#GOPATH) which holds all go modules & build artifacts?

**Review by arajkumar — COMMENTED** [review]
Overall changes looks ok to me. Could you please add a test to cover newly added code?

**Comment by dgpatelgit** [comment]
For refactoring activity, new story https://issues.redhat.com/browse/APPAI-1477 is created.

**Comment by arajkumar** [comment]
[test]

**Review by arajkumar — CHANGES_REQUESTED** [review]
Overall changes looks good to me.

@dgpatelgit Could you please update README.md.

**Review by arajkumar — APPROVED** [review]
lgtm

**Comment by arajkumar** [comment]
[test]

**Comment by dgpatelgit** [comment]
> [test]

I have added test for generating gograph.txt manifest. Rest of the flow is common as other ecosystem. 
Do you see any specific test to be added?

**Review by mathur07 — APPROVED** [review]

**Pull request merged**